parser/src/parser.cpp

```diff
@@ -225,24 +250,64 @@ int main(int argc, char* argv[])
   if (vm.count("force"))
     cc::util::removeTables(db, SQL_DIR);
 
-  cc::util::createTables(db, SQL_DIR);
+  if (vm.count("force") || isNewDb)
+    cc::util::createTables(db, SQL_DIR);
 
   //--- Start parsers ---//
 
   cc::parser::SourceManager srcMgr(db);
   cc::parser::ParserContext ctx(db, srcMgr, compassRoot, vm);
   pHandler.createPlugins(ctx);
 
+  // TODO: Handle errors returned by preparse().
+  std::vector<std::string> topologicalOrder = pHandler.getTopologicalOrder();
+  for (auto it = topologicalOrder.rbegin(); it != topologicalOrder.rend(); ++it)
+  {
+    LOG(info) << "[" << *it << "] preparse started!";
+    if(!pHandler.getParser(*it)->preparse())
+    {
+      LOG(error) << "[" << *it << "] preparse failed!";
+      return 2;
+    }
+  }
+
+  // TODO: Consider whether there is a better place for this code.
+  (cc::util::OdbTransaction(ctx.db))([&]
+  {
+    for (auto& item : ctx.fileStatus)
+    {
+      switch (item.second)
+      {
+        case cc::parser::IncrementalStatus::MODIFIED:
+        case cc::parser::IncrementalStatus::DELETED:
+        {
+          LOG(info) << "Database cleanup: " << item.first;
+
+          // Fetch file from SourceManager by path
```

Comment:
No dot at the end of sentence.

Reply:
You are joking now, right? Aside that requesting a change at a such a low level (comment style) is comedic; single line, short comments nowhere end in punctuation dot in the CodeCompass codebase last time I checked.

Reply:
Style-based fixup requests are common, see the comment above about what usually happens in the LLVM community too.

Reply:
There is no fixup, your suggestion is **against** the applied coding style convention of the project.